src/pages/VMView.vue

```diff
@@ -76,26 +95,31 @@ export default {
       this.loading = true
       var params = {
         addresses: [this.account.address],
-        pagination: 1000,
+        pagination: this.pagination.per_page,
+        page: this.current_page,
         message_type: 'PROGRAM'
       }
       if (this.search.length > 0) {
         params.hashes = [this.search]
       }
       await messages.get_messages(params).then(async (response) => {
+        this.pagination.total = response.pagination_total
+        this.pagination.pages = Math.ceil(response.pagination_total / this.pagination.per_page)
         var programsTmp = response.messages
         for (var i = 0; i < programsTmp.length; i++) {
-          let tx = programsTmp[i].content.code.ref
+          let tx = programsTmp[i].content?.code.ref
           // retrieve store messages
-          await messages.get_messages({
-            addresses: [this.account.address],
-            hashes: [tx]
-          }).then(async (response) => {
-            let storeObj = response.messages[0].content
-            programsTmp[i].storeObj = storeObj
-          }).catch(() => {
-            this.loading = false
-          })
+          if (tx & !programsTmp[i].forgotten_by) {
+            await messages.get_messages({
+              addresses: [this.account.address],
+              hashes: [tx]
+            }).then(async (response) => {
```

Comment:
Please don't use `await` AND `then` constructs in the same function. The await has no effect in this specific case, this is the same for all the other "await + then" you have in this page.

Reply:
You right, I removed it!